### ISSUE TITLE
List, Tree: Make sure sub-pixel antialiasing works across all lists and trees

### ISSUE THREAD

**Issue by alexandrudima** [microsoft/vscode#84715]
@joaomoreno says:

**TL;DR** Are you a list/tree user? You need to provide a background color to the list/tree in order to preserve sub-pixel anti-aliasing in Windows and Linux.

Recent findings by @alexandrudima and @bpasero uncovered that sub-pixel anti-aliasing gets disabled when a layer is transparent. The list/tree automatically creates a new rendering layer for scrolling performance. So, in order to preserve sub-pixel AA, the root DOM element of the layer must have a background color. That background color depends on its placement in the workbench.

**Example:** the Explorer should just forward the sidebar background color:

https://github.com/microsoft/vscode/blob/90a0eb3a145b7c8b8f548a6183475291ace63719/src/vs/workbench/contrib/files/browser/views/explorerView.ts#L387

**Note:** Only opaque colors work here, since passing in a transparent color would defeat the whole purpose and would potentially place the same transparent background color twice, making it darker. If a theme or user eventually passes in a transparent color, the list/tree will ignore it and print the following warning to the devtools:

> List with id 'list_id_4' was styled with a non-opaque background color. This will break sub-pixel antialiasing.

**Note 2:** macOS doesn't have sub-pixel AA. This only affects Linux and Windows.

---

**Workbench List/Tree Users**

- [x] `referencesWidget.ts` @jrieken 
- [x] `breadcrumbsPicker.ts` @jrieken 
- [x] `notificationsList.ts` @bpasero 
- [x] `quickInputList.ts` @chrmarti 
- [ ] `customView.ts` @alexr00 
- [x] `callHierarchyPeek.ts` @jrieken  
- [x] `commentsTreeViewer.ts` @rebornix 
- [x] `breakpointsView.ts` @isidorn 
- [x] `callStackView.ts` @isidorn 
- [x] `debugHover.ts` @isidorn 
- [x] `loadedScriptsView.ts` @isidorn 
- [x] `repl.ts` @isidorn 
- [x] `variablesView.ts` @isidorn 
- [x] `watchExpressionsView.ts` @isidorn 
- [x] `extensionsViewer.ts` @sandy081 
- [x] `extensionsViews.ts` @sandy081 
- [x] `runtimeExtensionsEditor.ts` @sandy081 
- [x] `explorerView.ts` @joaomoreno
- [x] `openEditorsView.ts` @isidorn 
- [x] `markersPanel.ts` @sandy081 
- [x] `outlinePanel.ts` @jrieken 
- [x] `keybindingsEditor.ts` @sandy081 
- [x] `mainPanel.ts` @joaomoreno 
- [x] `repositoryPanel.ts` @joaomoreno 
- [x] `searchView.ts` @roblourens 

**Non-Workbench List/Tree Users**

These trees do not benefit from the workbench facilities. They use their own stylers. The `listBackground` needs to be added to those styles. 

- [x] `tocTree.ts` @roblourens 
- [x] `settingsTree.ts` @roblourens 

---

**Good: Sub-pixel AA**

![image](https://user-images.githubusercontent.com/22350/69048851-c6ceb480-09fe-11ea-8ca8-2afec7e14bef.png)

**Bad: Normal AA**

![image](https://user-images.githubusercontent.com/22350/69048864-cdf5c280-09fe-11ea-9b21-392db171107b.png)

---

<details>
<summary>
Original issue by @alexandrudima :
</summary>
fyi @bpasero 

* take a screenshot on Windows of a tree, like the Explorer
* zoom a lot
* observe it uses greyscale text rendering:
![image](https://user-images.githubusercontent.com/5047891/68785153-059aee00-063e-11ea-9b4e-45763d37043e.png)

* if I make the following change in `listView.ts`:
![image](https://user-images.githubusercontent.com/5047891/68785262-2f541500-063e-11ea-8e5e-ba8b45f21256.png)

* then, the text renders with subpixel AA:
![image](https://user-images.githubusercontent.com/5047891/68785346-56124b80-063e-11ea-9e22-c9515e97bdf7.png)

*TL;DR*: From what I could tell, as soon as you have a separate layer, the layer dom node itself must have an opaque background, at all times. If it gets transparent even for a split second, it will permanently render with grayscale text. More of my findings documented [here](https://bugs.chromium.org/p/chromium/issues/detail?id=1016062#c31)

</details>

**Comment by joaomoreno** [comment]
This needs to be set as a non-optional tree configuration and needs adoption from every tree user.

**Comment by isidorn** [comment]
Thanks a lot for pushing on this.
I have adopted it across my trees.

**Comment by isidorn** [comment]
These changes actually seem to break the tree selection.
When I run out of source the blue tree selection is no longer rendered.
For example just start vscode and click on the explorer.

**Comment by joaomoreno** [comment]
@isidorn Pushed fix for it.

**Comment by bpasero** [comment]
I adopted for notifications, however requires more work because notifications use drop shadows. I opened https://github.com/microsoft/vscode/issues/85043 as follow up.

**Comment by sandy081** [comment]
✅

**Comment by dannymcgee** [comment]
Apologies for barging in — I'm currently not seeing subpixel AA in the main editor window as of 1.40.0 — I immediately noticed the difference in perceived text coloring after updating (colors appear much less saturated with grayscale AA). Is this the same or a related issue, or should I post a separate one?

FWIW, I added `* { background: #000 !important; }` in devtools as a test, but did not see subpixel AA flip on as I would have expected based on the description — maybe there's something I'm missing or not understanding?

**Comment by rebornix** [comment]
✅

**Comment by joaomoreno** [comment]
@roblourens I've fixed search and settings. Create this additional one for settings: https://github.com/microsoft/vscode/issues/85106

**Comment by joaomoreno** [comment]
@alexr00 I've fixed custom views: it was an easy one.

**Comment by Tyriar** [comment]
I'm verifying now, I'll create new more targeted issues for any parts that aren't rendering correctly since this issue contains so much
